File: ViewBase/Source/Converters/HiddenVisibilityToBoolConverter.vb
Imports System.Globalization
Imports System.Windows
Imports System.Windows.Data

Public Class HiddenVisibilityToBoolConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        If value Is Nothing Then
            Return False
        End If
        If value.GetType Is GetType(Visibility) Then
            Dim vis = CType(value, Visibility)
            Return If(vis = Visibility.Visible, True, False)
        End If
        Return False
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        If value Is Nothing Then
            Return Visibility.Hidden
        End If
        If value.GetType Is GetType(Boolean) Then
            Dim bool = CType(value, Boolean)
            Return If(bool, Visibility.Visible, Visibility.Hidden)
        End If
        Return Visibility.Hidden
    End Function
End Class

File: BaseView/src/Converters/HiddenVisibilityToBoolConverter.cs
using System;
using System.Globalization;
using System.Windows;
using System.Windows.Data;

namespace Wells.BaseView.Converters
{
    public class HiddenVisibilityToBoolConverter : IValueConverter
    {
        object IValueConverter.Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value == null)
                return false;
            if (value.GetType() == typeof(Visibility))
                return (Visibility)value == Visibility.Visible ? true : false;
            return false;
        }

        object IValueConverter.ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value == null)
                return Visibility.Hidden;
            if (value.GetType() == typeof(bool))
                return (bool)value ? Visibility.Visible : Visibility.Hidden;
            return Visibility.Hidden;
        }
    }
}
